src/Parser/DeathKnight/Blood/Modules/Abilities.js

```diff
@@ -265,21 +265,30 @@ class Abilities extends CoreAbilities {
       {
         spell: SPELLS.RUNE_1,
         category: Abilities.SPELL_CATEGORIES.HIDDEN,
-        cooldown: haste => 10 / (1 + haste),
+        cooldown: (haste, combatantCurrent) => {
+          const multiplier = combatantCurrent.hasBuff(SPELLS.RUNE_MASTER.id) ? 0.4 : 0;
+          return 10 / (1 + haste) / (1 + multiplier);
+        },
         charges: 2,
       },
 
       {
         spell: SPELLS.RUNE_2,
         category: Abilities.SPELL_CATEGORIES.HIDDEN,
-        cooldown: haste => 10 / (1 + haste),
+        cooldown: (haste, combatantCurrent) => {
+          const multiplier = combatantCurrent.hasBuff(SPELLS.RUNE_MASTER.id) ? 0.4 : 0;
```

Comment:
Should the runic corruption bit be handled here too?

Reply:
Runic Corruption is Unholy only and Rune Master is blood only

Reply:
Oh ok, missed that too.

I was debugging all day at my job, I don't think I can look at code any more today ☹️